# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformMac.cpp
# Commit: c03c7df7cd61
# Full Hash: c03c7df7cd61c88de130f49dbceaa8a62c9a7758
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-05-14 21:58:21
# Regressor Bug: 1708768
# File Overlap Count: 2
# Description:
#   Bug 1708768 - Preliminary cleanup of gfxPlatformFontList creation/initialization code; no functional change. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114863
# ==============================================================================

diff -r 76dd6379cc01 -r c03c7df7cd61 gfx/thebes/gfxPlatformMac.cpp
--- a/gfx/thebes/gfxPlatformMac.cpp	Fri May 14 17:45:16 2021 +0000
+++ b/gfx/thebes/gfxPlatformMac.cpp	Fri May 14 18:05:44 2021 +0000
@@ -238,13 +238,8 @@
 
 bool gfxPlatformMac::ContentUsesTiling() const { return UsesTiling(); }
 
-gfxPlatformFontList* gfxPlatformMac::CreatePlatformFontList() {
-  gfxPlatformFontList* list = new gfxMacPlatformFontList();
-  if (NS_SUCCEEDED(list->InitFontList())) {
-    return list;
-  }
-  gfxPlatformFontList::Shutdown();
-  return nullptr;
+bool gfxPlatformMac::CreatePlatformFontList() {
+  return gfxPlatformFontList::Initialize(new gfxMacPlatformFontList);
 }
 
 void gfxPlatformMac::ReadSystemFontList(SystemFontList* aFontList) {